### PR TITLE
server: report server address and port

### PR DESCRIPTION
**Pull request by sam-github**
connected to https://github.com/strongloop-internal/scrum-nodeops/issues/1159

### CODE REVIEW

**Comment by sam-github** [comment]
Ryan, thoughts? This is going to be the lab app... I extended with server address so that the fact that the load balancer is routing to various instances is observable.

File: index.js [hunk 1]
```diff
@@ -10,8 +10,12 @@ module.exports = function() {
   });
 
   app.get('/api', function(req, res) {
+    var address = req.connection.address();
```
Comment (rmg) [inline index.js]: I'm not familiar with this API. Where does `req.connection` come from?

File: index.js [hunk 2]
```diff
@@ -10,8 +10,12 @@ module.exports = function() {
...
     res.json({
       message: 'Hello World',
+      pid: process.pid,
+      address: address.address,
```
Comment (rmg) [inline index.js]: I expected to see `req.socket.localAddress` here. Is that not what is desired?

File: index.js [hunk 3]
```diff
@@ -10,8 +10,12 @@ module.exports = function() {
...
+      port: address.port,
```
Comment (rmg) [inline index.js]: Ditto here, with `req.socket.localPort`

File: index.js (hunk 1 above)
Comment (sam-github) [inline index.js]: not sure, maybe undocumented express or node api? I'll use your suggestion.

**Comment by sam-github** [comment]
@cgole wants to use a  browser to hit the app, so host info should be in the / route as well as the /api route.

**Comment by rmg** [comment]
Should `res.send()` be `res.end()`? Pretty minor either way, so LGTM.

**Comment by sam-github** [comment]
@slnode test please

**Comment by sam-github** [comment]
@slnode test please

**Pull request merged**